curl.sh

```diff
@@ -1,10 +1,41 @@
 package: curl
-version: "1.0"
-system_requirement_missing: |
-  curl and its development package are missing from your system.
-   * RHEL-compatible systems: you will probably need "curl" and "curl-devel" packages.
-   * Ubuntu-compatible systems: you will probably need "curl" and "libcurl4-openssl-dev" (or "libcurl4-gnutls-dev").
-system_requirement: ".*"
-system_requirement_check: |
-  curl --version > /dev/null; if test $? = 127; then exit 1; else printf "#include <curl/curl.h>\nint main() {}\n" | cc -xc -lcurl - -o /dev/null || exit 2; fi; exit 0
+version: "7.70.0"
+tag: curl-7_70_0
+source: https://github.com/curl/curl.git
+build_requires:
+ - "OpenSSL:(?!osx)"
+ - CMake
 ---
+#!/bin/bash -e
+
+if [[ $ARCHITECTURE = osx* ]]; then
+  OPENSSL_ROOT=$(brew --prefix openssl)
+else
+  ${OPENSSL_ROOT:+env LDFLAGS=-Wl,-R$OPENSSL_ROOT/lib}
+fi
+
+rsync -av --delete --exclude="**/.git" $SOURCEDIR/ .
+
+./buildconf
+./configure --prefix=$INSTALLROOT ${OPENSSL_ROOT:+--with-ssl=$OPENSSL_ROOT} --disable-static
+make ${JOBS:+-j$JOBS}
+make install
+
+# Modulefile
+mkdir -p etc/modulefiles
+cat > etc/modulefiles/$PKGNAME <<EoF
+#%Module1.0
+proc ModulesHelp { } {
+  global version
+  puts stderr "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
+}
+set version $PKGVERSION-@@PKGREVISION@$PKGHASH@@
+module-whatis "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
+# Dependencies
+module load BASE/1.0 ${OPENSSL_REVISION:+OpenSSL/$OPENSSL_VERSION-$OPENSSL_REVISION}
+# Our environment
+prepend-path LD_LIBRARY_PATH \$CURL_ROOT/lib
```

Comment:
@yuw726 You are missing `set CURL_ROOT \$::env(BASEDIR)/$PKGNAME/\$version` before defining `LD_LIBRARY_PATH`.

Reply:
or you could use the automated generation of modulefiles... See the alibuild reference docs.

Reply:
Thank you for the comments, I've added corresponding lines.